### PR TITLE
    [ci] remove old bazel flags, and add debug flavor

### PR DESCRIPTION
**Pull request by aslonnie**
add linux-debug / windows-debug / macos-debug for debug usage

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -71,17 +71,6 @@ build:windows --color=yes
 build:clang-cl --per_file_copt="-\\.(asm|S)$@-fansi-escape-codes"
 build:clang-cl --per_file_copt="-\\.(asm|S)$@-fcolor-diagnostics"
 
-build:manylinux2010 --copt="-Wno-unused-result"
-build:manylinux2010 --linkopt="-lrt"
-
-# Debug build flags. Uncomment in '-c dbg' builds to enable checks in the C++ standard library:
```
Comment (pcmoritz) [inline .bazelrc:77]: I think we should keep those debug flags, I have found them very useful in the past when debugging some crazy STL corruptions :)

Comment (aslonnie) [reply .bazelrc:77]: is it possible to move these to some docs or something, rather than putting them in default `.bazelrc` in comment form?

**Review by krfricke — APPROVED** [review]
Fine with me but will leave the comment/docs decision up to @pcmoritz

**Comment by can-anyscale** [comment]
clean up my queue, feel free to add me back if need to

File: .bazelrc (hunk 1 above)
Comment (aslonnie) [reply .bazelrc:85]: adding `*-debug` flavors for the debug flags.

**Comment by aslonnie** [comment]
reviving this.. respecting @pcmoritz 's opinion, keeping the flags, but not in comments.

**Comment by aslonnie** [comment]
merging this.. if people do not like this, we can always revert.

**Pull request merged**